crates/bitwarden-crypto/src/keys/master_key.rs

```diff
@@ -269,28 +284,32 @@ mod tests {
     fn test_make_user_key() {
         let mut rng = rand_chacha::ChaCha8Rng::from_seed([0u8; 32]);
 
-        let master_key = MasterKey(SymmetricCryptoKey::new(
-            Box::pin(
+        let master_key = MasterKey(KdfDerivedKeymaterial {
+            key_material: Box::pin(
                 [
                     31, 79, 104, 226, 150, 71, 177, 90, 194, 80, 172, 209, 17, 129, 132, 81, 138,
                     167, 69, 167, 254, 149, 2, 27, 39, 197, 64, 42, 22, 195, 86, 75,
                 ]
                 .into(),
             ),
-            None,
-        ));
+        });
 
         let (user_key, protected) = make_user_key(&mut rng, &master_key).unwrap();
+        let user_key_unwrapped = if let SymmetricCryptoKey::Aes256CbcHmacKey(k) = &user_key.0 {
+            k
+        } else {
+            panic!("User key is not an Aes256CbcHmacKey");
+        };
```

Comment:
You should also be able to do a let-else here as well, I think:

```suggestion
        let SymmetricCryptoKey::Aes256CbcHmacKey(user_key_unwrapped) = &user_key.0 else {
            panic!("User key is not an Aes256CbcHmacKey");
        };
```